drsclient/client.py

```diff
@@ -0,0 +1,303 @@
+        if asyncio._get_running_loop() is not None:
+            return coro
+        result = None
+        try:
+            while True:
+                result = coro.send(result)
+        except StopIteration as si:
+            return si.value
+
+    return _wrapper
+
+
+class SyncClient(httpx.Client):
+    async def __aenter__(self):
+        return self.__enter__()
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        self.__exit__(exc_type, exc_val, exc_tb)
+
+    async def request(self, *args, **kwargs):
+        return super().request(*args, **kwargs)
+
+
+def json_dumps(data):
+    return json.dumps({k: v for (k, v) in data.items() if v is not None})
+
+
+def timeout_wrapper(func):
+    def timeout(*args, **kwargs):
+        kwargs.setdefault("timeout", 60)
+        return func(*args, **kwargs)
+
+    return timeout
+
+
+def retry_and_timeout_wrapper(func):
+    def retry_logic_with_timeout(*args, **kwargs):
+        kwargs.setdefault("timeout", 60)
+        retries = 0
+        while retries < MAX_RETRIES:
+            try:
+                return func(*args, **kwargs)
+            except requests.exceptions.ReadTimeout:
+                retries += 1
+                if retries == MAX_RETRIES:
+                    raise
+
+    return retry_logic_with_timeout
+
+
+class DrsClient(object):
+    def __init__(self, baseurl, auth=None):
+        self.auth = auth
+        self.url = baseurl
+
+    def url_for(self, *path):
+        subpath = "/".join(path).lstrip("/")
+        return "{}/{}".format(self.url.rstrip("/"), subpath)
+
+    def check_status(self):
+        """Check that the API we are trying to communicate with is online"""
+        resp = httpx.get(self.url + "/index")
+        return resp
+
+    def get(self, guid, endpoint="/ga4gh/drs/v1/objects", expand=False):
+        params = {"expand": expand}
+        response = self._get(SyncClient, endpoint, guid, params=params)
+        return response
+
+    async def async_get(self, guid, endpoint="/ga4gh/drs/v1/objects", expand=False):
+        params = {"expand": expand}
+        response = await self._get(httpx.AsyncClient, endpoint, guid, params=params)
+        return response
+
+    def download(self, guid, protocol, endpoint="/ga4gh/drs/v1/objects"):
+        endpoint += "/" + guid + "/access/" + protocol
+        response = self._get(SyncClient, endpoint)
+        return response
+
+    async def async_download(self, guid, protocol, endpoint="/ga4gh/drs/v1/objects"):
+        endpoint += "/" + guid + "/access/" + protocol
+        response = await self._get(httpx.AsyncClient, endpoint)
+        return response
+
+    def get_all(
+        self,
+        endpoint="/ga4gh/drs/v1/objects",
+        start=None,
+        limit=None,
+        page=None,
+        form=None,
+    ):
+        """
+        Get a list of bundle, object or both.
+
+        Args:
+            endpoint (str): pick between '/ga4gh/drs/v1/objects', 'bundle' or 'index' endpoint.
```

Comment:
```suggestion
            endpoint (str): pick between '/ga4gh/drs/v1/objects', 'bundle' or 'index' endpoints (if calling a Gen3 Indexd instance).
```
again pretty indexd-specific but oh well